src/compiler/checker.ts

```diff
@@ -48243,12 +48283,24 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
     }
 
     function isExpandoFunctionDeclaration(node: Declaration): boolean {
-        const declaration = getParseTreeNode(node, isFunctionDeclaration);
+        const declaration = getParseTreeNode(node, (n): n is FunctionDeclaration | VariableDeclaration => isFunctionDeclaration(n) || isVariableDeclaration(n));
         if (!declaration) {
             return false;
         }
-        const symbol = getSymbolOfDeclaration(declaration);
-        if (!symbol || !(symbol.flags & SymbolFlags.Function)) {
+        let symbol: Symbol;
+        if (isVariableDeclaration(declaration)) {
+            if (declaration.type || !isVarConstLike(declaration)) {
+                return false;
+            }
+            if (!(declaration.initializer && isFunctionExpressionOrArrowFunction(declaration.initializer))) {
+                return false;
+            }
+            symbol = getSymbolOfDeclaration(declaration.initializer);

```

Comment:
This doesn't seem right, but I'd have to do some investigation to make sure. My memory is that 

```
const f = function g() { g.ex1 = 1 }
f.ex2 = 2
```

has to work in JS, and that `const f` has the symbol for it, not `function g`. But I might be misremembering.
I also thought there were utilities that give the right node to get the symbol from. I'll see what I can turn up.

Reply:
No, I'm wrong, according to `getExpandoSymbol` the initialiser is the one with all the members.

Reply:
So no action needed?